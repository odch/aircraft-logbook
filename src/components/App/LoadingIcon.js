import React from 'react'
import PropTypes from 'prop-types'
import CircularProgress from '@material-ui/core/CircularProgress'
import withStyles from '@material-ui/core/styles/withStyles'

const styles = () => ({
  container: {
    position: 'absolute',
    width: '100%',
    height: '100%'
  },
  icon: {
    position: 'absolute',
    top: '50%',
    left: '50%',
    margin: '-25px 0 0 -25px'
  }
})

const LoadingIcon = props => (
  <div className={props.classes.container}>
    <CircularProgress size={50} className={props.classes.icon} />
  </div>
)

LoadingIcon.propTypes = {
  classes: PropTypes.object.isRequired
}

export default withStyles(styles)(LoadingIcon)
