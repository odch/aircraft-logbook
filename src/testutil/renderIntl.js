import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
