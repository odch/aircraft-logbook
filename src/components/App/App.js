import React from 'react'
import PropTypes from 'prop-types'
import LoadingIcon from './LoadingIcon'

class App extends React.Component {
  componentDidMount() {
    this.props.watchAerodromes()
  }

  render() {
    if (!this.props.auth.isLoaded) {
      return <LoadingIcon />
    }
    return this.props.children
  }
}

App.propTypes = {
  auth: PropTypes.shape({
    isLoaded: PropTypes.bool.isRequired,
    isEmpty: PropTypes.bool.isRequired
  }),
  children: PropTypes.element,
  watchAerodromes: PropTypes.func.isRequired
}

export default App
