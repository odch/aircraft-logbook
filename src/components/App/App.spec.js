import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import App from './App'
import renderIntl from '../../testutil/renderIntl'

describe('components', () => {
  describe('App', () => {
    it('renders loading icon if not initialized', () => {
      const auth = {
        isLoaded: false,
        isEmpty: true
      }
      const store = configureStore()({
        firebase: {
          auth
        }
      })
      const tree = renderIntl(
        <Provider store={store}>
          <App auth={auth} />
        </Provider>
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders child if authenticated', () => {
      const auth = {
        isLoaded: true,
        isEmpty: false,
        email: 'test@example.com'
      }
      const store = configureStore()({
        firebase: {
          auth,
          profile: {}
        }
      })
      const tree = renderIntl(
        <Provider store={store}>
          <App auth={auth}>
            <div>content</div>
          </App>
        </Provider>
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
