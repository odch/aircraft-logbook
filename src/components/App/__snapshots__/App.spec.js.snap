// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components App renders app if authenticated 1`] = `
Array [
  <div
    data-cy="username"
  >
    test@example.com
  </div>,
  <button
    onClick={[Function]}
  >
    <span>
      Abmelden
    </span>
  </button>,
]
`;

exports[`components App renders login form if not authenticated 1`] = `
<main
  className="LoginForm-layout-1"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root-3 MuiFormControl-marginNormal-4 MuiFormControl-fullWidth-6"
    >
      <label
        className="MuiFormLabel-root-14 MuiFormLabel-required-19 MuiInputLabel-root-7 MuiInputLabel-formControl-8 MuiInputLabel-animated-11"
        data-shrink={false}
        htmlFor="email"
      >
        <span>
          E-Mail
        </span>
        <span
          className="MuiFormLabel-asterisk-20"
        >
           *
        </span>
      </label>
      <div
        className="MuiInputBase-root-34 MuiInput-root-21 MuiInput-underline-25 MuiInputBase-formControl-35 MuiInput-formControl-22"
        data-cy="email"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="email"
          autoFocus={true}
          className="MuiInputBase-input-44 MuiInput-input-29"
          disabled={false}
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root-3 MuiFormControl-marginNormal-4 MuiFormControl-fullWidth-6"
    >
      <label
        className="MuiFormLabel-root-14 MuiFormLabel-required-19 MuiInputLabel-root-7 MuiInputLabel-formControl-8 MuiInputLabel-animated-11"
        data-shrink={false}
        htmlFor="password"
      >
        <span>
          Passwort
        </span>
        <span
          className="MuiFormLabel-asterisk-20"
        >
           *
        </span>
      </label>
      <div
        className="MuiInputBase-root-34 MuiInput-root-21 MuiInput-underline-25 MuiInputBase-formControl-35 MuiInput-formControl-22"
        data-cy="password"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="current-password"
          className="MuiInputBase-input-44 MuiInput-input-29 MuiInputBase-inputType-47 MuiInput-inputType-32"
          disabled={false}
          id="password"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
          value=""
        />
      </div>
    </div>
    <button
      className="MuiButtonBase-root-77 MuiButton-root-51 MuiButton-contained-62 MuiButton-containedPrimary-63 MuiButton-raised-65 MuiButton-raisedPrimary-66 MuiButton-fullWidth-76"
      data-cy="submit"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="submit"
    >
      <span
        className="MuiButton-label-52"
      >
        <span>
          Anmelden
        </span>
      </span>
      <span
        className="MuiTouchRipple-root-80"
      />
    </button>
  </form>
</main>
`;
